extensions/eclipse-che-theia-plugin-remote/src/node/plugin-remote.ts

```diff
@@ -20,6 +20,38 @@ process.on('SIGINT', () => {
     process.exit();
 });
 
+process.on('uncaughtException', (err: Error) => {
+    console.error('Remote plugin node: got an uncaught exception', err);
+});
+
+// tslint:disable-next-line: no-any
+const unhandledPromises: Promise<any>[] = [];
+
+// tslint:disable-next-line: no-any
+process.on('unhandledRejection', (reason: any, promise: Promise<any>) => {
+    unhandledPromises.push(promise);
+    setTimeout(() => {
+        const index = unhandledPromises.indexOf(promise);
+        if (index >= 0) {
+            promise.catch(err => {
+                unhandledPromises.splice(index, 1);
+                console.error(`Remote plugin node: promise rejection not handled in one second: ${err}`);
```

Comment:
I think we can remove the 'node' word there as it can be confusing
We could add container name

Reply:
ok